catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/WhenBrickTest.java

```diff
@@ -0,0 +1,167 @@
+import org.catrobat.catroid.physics.content.bricks.CollisionReceiverBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickCoordinatesProvider;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.matchers.BrickCategoryListMatchers;
+import org.catrobat.catroid.uiespresso.util.matchers.BrickPrototypeListMatchers;
+import org.catrobat.catroid.uiespresso.util.matchers.ScriptListMatchers;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.util.matchers.rtl.RtlViewDirection.isViewRtl;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+@RunWith(AndroidJUnit4.class)
+public class WhenBrickTest {
+	List<Integer> whenBrickList = new ArrayList<Integer>();
+	Map<Integer, Class<?>> whenBrickMap = new HashMap<Integer, Class<?>>();
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+
+		Script script = BrickTestUtils.createProjectAndGetStartScript("WhenBrickTest");
+		script.addBrick(new PlaceAtBrick());
+
+		whenBrickMap.put(R.string.brick_when_started,WhenStartedBrick.class);
+		whenBrickMap.put(R.string.brick_when_becomes_true,WhenConditionBrick.class);
+		whenBrickMap.put(R.string.brick_when_cloned,WhenClonedBrick.class);
+		whenBrickMap.put(R.string.brick_when_background,WhenBackgroundChangesBrick.class);
+		whenBrickMap.put(R.string.brick_when_touched,WhenTouchDownBrick.class);
+		whenBrickMap.put(R.string.brick_broadcast_receive,BroadcastReceiverBrick.class);
+		whenBrickMap.put(R.string.brick_collision_receive,CollisionReceiverBrick.class);
+
+		whenBrickList.add(R.string.brick_when_started);
```

Comment:
duplication, theres no need for this list, to iterate through the dictionary, you can use whenBrickMap.keySet() which returns a set of all contained keys..